src/api/app/controllers/webui/request_controller.rb

```diff
@@ -340,12 +340,18 @@ def set_filter_involvement
     @filter_involvement = 'all' if ALLOWED_INVOLVEMENTS.exclude?(@filter_involvement)
   end
 
+  def set_filter_action_type
+    @filter_action_type = params[:action_type].presence || []
+    @filter_action_type = @filter_action_type.intersection(BsRequestAction::TYPES)
+  end
+
   def filter_requests
     @bs_requests = filter_by_involvement(@bs_requests, @filter_involvement)
+    @bs_requests = @bs_requests.for_action_type(@filter_action_type) if @filter_action_type.present?
```

Comment:
```suggestion
    @bs_requests = @bs_requests.bs_request_actions.where(type: @filter_action_type) if @filter_action_type.present?
```

Reply:
Or better yet, just make this a scope on the `bs_request_actions` association in `BsRequest`

Reply:
```
has_many :blah do
  def with_types(types: [])
    where(type: types)
  end
end
```

Reply:
How is this more complicated? This is how you scope associations the Rails way.

Reply:
doing `.where...` is just the "not-shared" way. The scope on the association people will find if they are looking for a way to get actions by type...

Reply:
Ah no this is bullshit, we are not filtering the `BsRequestActions` association but `BsRequest`... sorry for the noise.